### PR TITLE
OSD-16136 : In-built SOP Feature Added

### PR DESCRIPTION
**Pull request by devppratik**
### What type of PR is this?

_feature_

### What this PR does / Why we need it?
This PR adds the feature to view SOP related to alerts inside the kite interface itself
Pressing S on the alerts page, will open up the SOP on a new tab. 

image

### Which Jira/Github issue(s) does this PR fix?

_Resolves #[OSD-16136](https://issues.redhat.com//browse/OSD-16136)_

### Pre-checks (if applicable)

- [ ] Ran unit tests locally against the changes
- [ ] Included documentation changes with PR

### CODE REVIEW

File: cmd/kite/login/cmd.go [hunk 1]
```diff
@@ -92,6 +100,17 @@ func loginHandler(cmd *cobra.Command, args []string) error {
...
+
+	// API key is not found in the config file
+	if len(cfg.AccessToken) == 0 {
+
+		// Create a new API key and store it in the config file
+		err = generateNewAccessToken(cfg)
+
+		if err != nil {
+			return err
+		}
```
Comment (supreeth7) [inline cmd/kite/login/cmd.go:113]:
```suggestion
		if err = generateNewAccessToken(cfg); err != nil {
		return err
		}
```

File: cmd/kite/login/cmd.go [hunk 2]
```diff
@@ -159,6 +178,26 @@ func generateNewKey(cfg *config.Config) (err error) {
 	return nil
 }
 
+// generateNewKey prompts the user to create a new API key and saves it to the config file.
```
Comment (supreeth7) [inline cmd/kite/login/cmd.go:181]:
```suggestion
// generateNewAccessToken prompts the user to create a new API key and saves it to the config file.
```

**Review by supreeth7 — CHANGES_REQUESTED** [review]

File: cmd/kite/login/cmd.go (hunk 1 above)
Comment (devppratik) [reply cmd/kite/login/cmd.go:113]: Ack

**Review by supreeth7 — CHANGES_REQUESTED** [review]
Solid work on this!

We _really_ need to change the `login_test`, didn't realize it was all fairy dust without meaningful edge cases until I took a pass on it today. 

Please create a Jira to specifically **rewrite** all the `login` tests as part of M4.

File: cmd/kite/login/cmd.go [hunk 3]
```diff
@@ -51,6 +53,12 @@ func init() {
 		"",
 		"Access API key/token generated from "+constants.APIKeyURL+"\nUse this option to overwrite the existing API key.",
 	)
+	Cmd.Flags().StringVar(
+		&loginArgs.accessToken,
+		"access-token",
+		"",
+		"GitHub Personal Access Token generated from "+constants.AccessTokenURL+"\nUse this option to overwrite the existing Access Token.",
```
[the remark this answers is not in the document]
Comment (supreeth7) [reply cmd/kite/login/cmd.go:60]: Nvm, this can be addressed in another PR.

**Comment by supreeth7** [comment]
Overriding CI to go over faulty tests for now.

/override ci/prow/coverage
/override ci/prow/test

**Comment by supreeth7** [comment]
Please comment out the entire `login_test.go` file and add a `// TODO` header so it doesn't block other PRs

**Review by supreeth7 — CHANGES_REQUESTED** [review]
@devppratik can you rebase?

**Comment by devppratik** [comment]
@supreeth7 I have rebased the PR

**Review by supreeth7 — APPROVED** [review]
/lgtm
